### PR TITLE
Test messages

### LINKED ISSUE #84: Add tests for each messages class
As seen in the other test files
Test each class:
Messages
MessageBoard
MessageTags

add functions to make sure the models are correct.

### PR DESCRIPTION
**Pull request by shakedhagag**
## Test files for each messages model class

added 3 files:
1. test_messages
2. test_msgboards
3. test_msgtags

closes #84

### CODE REVIEW

File: AcadeMeData/test_messages.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    message.save()
+    return message
+
+
+@pytest.mark.django_db
+class TestMessagesModel:
+
+    def test_get_msg(self, generate_message):
+        msg = generate_message
+        msg_test = Messages.get_msg_by_id(1)
```
Comment (Yarboa) [inline AcadeMeData/test_messages.py:27]: I do not think this method helps in finding messages.
Did you think what are the keys for the messages

File: AcadeMeData/test_msgboard.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    msgboard = MessageBoards(id=id, courseName=courseName)
+    msgboard.save()
+    return msgboard
+
+
+@pytest.mark.django_db
+class TestMessageBoardModel:
+    @pytest.fixture
+    def generate_msgboard(self, id=1, courseName="Linear Algebra"):
+        msgboard = MessageBoards(id=id, courseName=courseName)
```
Comment (Yarboa) [inline AcadeMeData/test_msgboard.py:16]: This is a fixture, should go out side

Where is the test of message boards? what do you want to test?

File: AcadeMeData/test_msgtags.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    user_data = {'username': "username2212", 'password': "password", 'email': "user@example.com", 'type': "S",
+                 'university': "RU",
+                 'degree': "CS"}
+    user = User.create_user(*user_data)
+    return user
+
+
+@pytest.mark.django_db
+class TestMessageTagsModel:
+    def test_get_msg_tag(self, user_example, id=1, msgID=None):
```
Comment (Yarboa) [inline AcadeMeData/test_msgtags.py:16]: You should call fixtures here 
the test should call the create_message fixture that calls create_user  
Once you have a tag, you should search it and receive it, where is the test? i mean, if there is a tag_id you search, you have to verify the same user created, isnt it?

I mean if you have message tag

**Comment by Omeramsc** [comment]
That's a good example for a PR that could benefit from a good and thorough description. 

I read "test messages" and was sure this PR adds a test to some module
and it does, but it also adds the module itself and its migrations.

The linked PR also looks like it is relevant to tests.

File: AcadeMeData/test_messages.py (hunk 1 above)
Comment (shakedhagag) [reply AcadeMeData/test_messages.py:27]: This method is meant to test the get_msg_by_id function. 
I am initializing a message through a fixture with id=1
then I am inserting that message by id to another variable using the get_msg function.
then asserting if those 2 are the same and asserting if test_msg is a messages instance (object)
How is that not a good test for this specific function?

File: AcadeMeData/test_msgboard.py (hunk 2 above)
Comment (shakedhagag) [reply AcadeMeData/test_msgboard.py:16]: For some reason the test code disappeared will fix..

File: AcadeMeData/models.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
-from django.db import models
+from django.db import models, transaction
 from django.conf import settings
```
Comment (Yarboa) [inline AcadeMeData/models.py:2]: please remove it is main through PR#83
I hope you understand that you had to build this PR based on PR #83

File: AcadeMeData/migrations/0001_initial.py [hunk 5]
```diff
@@ -1,12 +1,12 @@
-# Generated by Django 3.2.9 on 2021-12-01 16:07
+# Generated by Django 3.2.9 on 2021-12-08 17:34
```
Comment (Yarboa) [inline AcadeMeData/migrations/0001_initial.py:1]: please remove this file it in main now

File: AcadeMeData/migrations/0003_messageBoards_test_data.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+from django.db import migrations, transaction
+
```
Comment (Yarboa) [inline AcadeMeData/migrations/0003_messageBoards_test_data.py:2]: please remove it is main through PR#83
I hope you understand that you had to build this PR based on PR #83

File: AcadeMeData/migrations/0004_messages_test_data.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+from django.db import migrations, transaction
+
+
```
Comment (Yarboa) [inline AcadeMeData/migrations/0004_messages_test_data.py:3]: please remove it is main through PR#83
I hope you understand that you had to build this PR based on PR #83

File: AcadeMeData/migrations/0005_messageTags_test_data.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+from django.db import migrations, transaction
+
```
Comment (Yarboa) [inline AcadeMeData/migrations/0005_messageTags_test_data.py:2]: please remove it is main through PR#83
I hope you understand that you had to build this PR based on PR #83

**Comment by shakedhagag** [comment]
I have made a new PR and fixed every issue this one had.

**Pull request closed without merging**